hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -637,4 +645,43 @@ private SubType getSubType(
         }
         return DEFAULT;
     }
+
+    private void validateTopLevelAllowances(
```

Comment:
This method looks like a good candidate to put in a validator, where it can have a bunch of unit tests around it

Reply:
Agree with @mhess-swl . But not blocker for this PR merge